### PR TITLE
chore: `input` coerce function returns paths matching glob pattern in openapi generator

### PR DESCRIPTION
**Pull request by deekshas8**
The coerce function for `input` option in openapi-generator now returns array of paths matching glob pattern.

- [x] I know which base branch I chose for this PR, as the default branch is `v2-main` now, which is not for v3 development.
- [ ] If my change will be merged into the `main` branch (for v3), I've updated (V3-Upgrade-Guide.md)[./V3-Upgrade-Guide.md] in case my change has any implications for users updating to SDK v3

### CODE REVIEW

File: packages/generator-common/src/options-parser.ts [hunk 1]
```diff
@@ -110,6 +110,45 @@ export function resolveRequiredPath<GeneratorOptionsT>(
...
+}
+
+/**
+ * Recursively searches through a given input path and returns all file paths as a string array.
+ * @param input - the path to the input directory.
+ * @returns all file paths as a string array.
+ * @internal
+ */
+export function getInputFilePaths(input: string): string[] {
+  if (glob.hasMagic(input)) {
```
Comment (FrankEssenberger) [inline packages/generator-common/src/options-parser.ts:137]: [q]I assume the `hasMagic` means that it is a directory? Please add a comment.

File: packages/generator-common/src/options-parser.ts [hunk 2]
```diff
@@ -110,6 +110,45 @@ export function resolveRequiredPath<GeneratorOptionsT>(
...
+ * @returns Absolute path as a `string`.
+ */
+export function resolveInputGlob<GeneratorOptionsT>(
+  arg: string,
+  options: GeneratorOptionsT & { config?: string }
+): string[] {
+  const inputPath = options.config
+    ? resolve(dirname(options.config), arg.toString())
+    : resolve(arg.toString());
+  return getInputFilePaths(inputPath.split(sep).join(posix.sep));
```
Comment (FrankEssenberger) [inline packages/generator-common/src/options-parser.ts:127]: [pp] When I workd on the [glob](https://github.com/SAP/cloud-sdk-js/blob/2650859415f7bc54de39caa7bf9156700dc8a628/packages/generator-common/src/options-parser.ts#L87) resolving I used the `cwd` property of the glob. To merge before has the same effect - We should align the two ways. Either remove the `cwd` in the mentioned method or use it also here to have the code as consistent as possible.

**Review by FrankEssenberger — APPROVED** [review]
Please consider my comment on aligning the code.

File: packages/generator-common/src/options-parser.ts (hunk 2 above)
Comment (deekshas8) [reply packages/generator-common/src/options-parser.ts:127]: Removed the `cwd`. Also fixed the tests that were affected as a result

File: packages/generator-common/src/options-parser.ts (hunk 1 above)
Comment (deekshas8) [reply packages/generator-common/src/options-parser.ts:138]: Added a comment. `hasMagic` checks if pattern has special characters.

**Pull request merged**